File: rb/features.rb
# TextValidation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module TextValidationFeatures
  def self.make_feature(name)
    case name
    when "base"
      TextValidationBaseFeature.new
    when "ratelimit"
      TextValidationRatelimitFeature.new
    when "retry"
      TextValidationRetryFeature.new
    when "test"
      TextValidationTestFeature.new
    when "timeout"
      TextValidationTimeoutFeature.new
    else
      TextValidationBaseFeature.new
    end
  end
end

File: py/features.py
# TextValidation SDK feature factory

from feature.base_feature import TextValidationBaseFeature
from feature.test_feature import TextValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TextValidationBaseFeature(),
        "test": lambda: TextValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
